### DISCUSSION TITLE
Rethinking ease of reach for undo/redo buttons, more form factor + touch considerations

### DISCUSSION THREAD

**Discussion by astrosticks** [flxzt/rnote#101] [Ideas]
I think we can do better when it comes to the ease of reach for undo/redo buttons. There has already been some [previous discussion about the alignment of the undo/redo buttons](https://github.com/flxzt/rnote/issues/54#issuecomment-1030415393) but I think that just flipping which side of the screen they appear on doesn't go far enough.

On a larger touch layout (like a tablet) it might not be possible to reach the other side of the screen with your thumb, or it might be awkward to touch the top half of the screen with the thumb on the hand supporting the device.

If there are any other functions that deserve similar reconsideration, this thread might be a good place to bring them up.

**Comment by astrosticks** [comment]
![undo-float](https://user-images.githubusercontent.com/32315760/152709992-ba311ad9-d450-4f1e-9d9b-9f1dd69c1600.png)
I think an interesting solution to explore for this would be with floating buttons that overlay the canvas and can attach at any of the four corners of the screen.

**Comment by flxzt** [comment]
It would certainly solve the issue of the crowded headerbar. I don't know if it would even need to be free-floating, maybe we could even utilize something like libpanel ([demo](https://www.youtube.com/watch?v=NDrzUKSCOvU&t=1s)) as an overlay. There are no rust bindings available currently though, but the gtk-rs folks are very diligent so maybe with a little helping effort we could convince them to create some.

This could also be combined with what I had in mind with [action pads](https://github.com/flxzt/rnote/issues/18#issuecomment-1021347577).

**Comment by flxzt** [comment]
I moved the buttons from the headerbar onto the canvas as an overlay. Thanks for the suggestion, I think it works well visually! They still need to be made repositionable, but at least the headerbar is now less crowded.

![Bildschirmfoto vom 2022-05-01 15-33-10](https://user-images.githubusercontent.com/19841886/166148435-26abb8ba-ecca-497e-9d7e-1c3cc91f3ee6.png)
